salt/grains/core.py

```diff
@@ -2528,7 +2528,7 @@ def ip4_interfaces():
             if "address" in inet:
                 iface_ips.append(inet["address"])
         for secondary in ifaces[face].get("secondary", []):
-            if "address" in secondary:
+            if "address" in secondary and "type" in secondary and secondary["type"] == "inet":
```

Comment:
```
if "address" in secondary and secondary.get("type") == "inet"
```

Reply:
FWIW the `key in thing and thing[key]` approach is *slightly* faster for most cases. Obviously less idiomatic :shrug: